sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/PreReservationCannotChange.java

```diff
@@ -10,27 +11,36 @@
  * Validates that the pre-reserved-role of a Pod cannot change.
  */
 public class PreReservationCannotChange implements ConfigValidator<ServiceSpec> {
+
     @Override
     public Collection<ConfigValidationError> validate(Optional<ServiceSpec> oldConfig, ServiceSpec newConfig) {
         if (!oldConfig.isPresent()) {
             return Collections.emptyList();
         }
 
-        Map<String, PodSpec> newPods = newConfig.getPods().stream()
+        final Map<String, PodSpec> newPods = newConfig.getPods().stream()
                 .collect(Collectors.toMap(podSpec -> podSpec.getType(), podSpec -> podSpec));
 
-        List<ConfigValidationError> errors = new ArrayList<>();
-        for (PodSpec oldPod : oldConfig.get().getPods()) {
-            PodSpec newPod = newPods.get(oldPod.getType());
-            if (!newPod.getPreReservedRole().equals(oldPod.getPreReservedRole())) {
+        final List<ConfigValidationError> errors = new ArrayList<>();
+        for (final PodSpec oldPod : oldConfig.get().getPods()) {
+            final PodSpec newPod = newPods.get(oldPod.getType());
+
+            if (newPod == null) {
```

Comment:
Is this check needed in other validators?

Reply:
I have looked through the other validators, and they explicitly handle this case -- either by first checking if the map contains the key or checking for null on the returned value.